### DISCUSSION TITLE
Tooltip

### DISCUSSION THREAD

**Discussion by mnbob70** [Megabit/Blazorise#3448] [Q&A] [answered]
The Tooltip component does not stay within the borders of a Modal which makes it difficult to read. I had to add `<br>` elements in the Text to keep it within the boundaries. Perhaps that is by design.

Is there any way to remove the transparency of the Tooltip? I'd prefer a solid background because the transparency shows text behind the Tooltip which makes reading difficult.

**Comment by stsrki** [comment]
Can you try the Multiline parameter? eg. `<Tooltip Text="Hello tooltip" Multiline>`

**Comment by mnbob70** [comment]
Multiline mostly works. It creates 3 lines but the edge of the Tooltip is aligned with the edge of the Modal. It would look better with some margin which I can manage by using `<br>` elements to create 4 lines.

What about the transparency. Is there an inherited property I can use to modify it as I don't see a specific property for Tooltip?

**Comment by stsrki** [comment]
There is no property to make it transparent. You can try adjusting the CSS and tippy.js theme https://atomiks.github.io/tippyjs/v5/themes/

**Comment by mnbob70** [comment]
The issue is that it **is** transparent. I'm looking for some way to remove the background transparency and make it solid. If the tooltip shows up over other text it makes reading the tooltip more difficult
![img2](https://user-images.githubusercontent.com/9770874/153955281-275c4802-c6b5-428b-97f6-77af47b93d83.png)
In the image, the word Frequency can be seen behind the tooltip text.

**Comment by stsrki** [comment, marked as answer]
The same still applies. You can use CSS to adjust the colors.

```css
.tippy-tooltip.tomato-theme {
  background-color: tomato;
  color: yellow;
}
```

Or if you're already using Blazorise theming you can also set it like this

```cshtml
<Blazorise.ThemeProvider Theme="@theme">
    <Router AppAssembly="typeof(App).Assembly">
        <Found Context="routeData">
            <RouteView RouteData="routeData" DefaultLayout="typeof(MainLayout)" />
        </Found>
        <NotFound>
            <p>Sorry, there's nothing at this address.</p>
        </NotFound>
    </Router>
    <MessageAlert />
    <NotificationAlert />
    <PageProgressAlert />
</Blazorise.ThemeProvider>
@code{
    private Theme theme = new()
    {
        TooltipOptions = new ThemeTooltipOptions
        {
            BackgroundColor = "#22c8ce"
        }
    };
}
```

**Comment by mnbob70** [comment]
The following worked for me. Added to site.css:
```
.tippy-content {
	background-color: rgb(128,128,128);
	border-radius: 4px;
}
```
My suggestion is that you consider removing the background opacity from the blazorise theme. I'm not sure why you would want a tooltip with any transparency.
```
.tippy-box[data-theme~='blazorise'] {
    background-color: RGBA(var(--b-tooltip-background-color-r, 128), var(--b-tooltip-background-color-g, 128), var(--b-tooltip-background-color-b, 128), var(--b-tooltip-background-opacity, 0.9));
    color: var(--b-tooltip-color, #ffffff);
}
```
